File: heron/lc014.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        String result = "";
        if (strs == null || strs.length == 0) {
            return result;
        }
        for (int i=0; i < strs[0].length(); i++) {
            char ch = strs[0].charAt(i);
            boolean isEnd = false;
            for (int j=1; j < strs.length; j++) {
                String s = strs[j];
                if (i >= s.length() || s.charAt(i) != ch) {
                    isEnd = true;
                }
            }
            if(!isEnd) {
                result += ch;
            } else {
                break;
            }
        }
        return result;
    }
}

File: heron/lc014.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """

        # check if the input is empty
        result = ""
        if strs is None or len(strs) == 0:
            return result

        # start to scan, pick the first one to act as base string
        s = strs[0]

        # walk through all char in the bast string
        for i in range(len(s)):
            # check with all other strings
            for j in range(len(strs)):
                if i >= len(strs[j]) or s[i] != strs[j][i]:
                    return result
            result += s[i]

        return result
